components/navigation/DocsNavItem.tsx

```diff
@@ -0,0 +1,81 @@
+import Link from 'next/link';
+
+export interface DocsNavItemProps {
+  title: string;
+  slug: string;
+  href?: string;
+  activeSlug: string;
+  sectionSlug?: string;
+  onClick?: () => void;
+  defaultClassName?: string;
+  inactiveClassName?: string;
+  activeClassName?: string;
+  bucket?: {
+      className: string;
+      icon: React.ComponentType<any>;
+  };
```

Comment:
Can we have same definition of bucket as defined in other files. I think there should be centralised definition of Bucket for all the buckets variable passed in the components.

Reply:
I tried that, but it showed errors when I used the same definition. That's why I made some props optional for buckets in some files and made some props compulsory for other files. Consequently, I created 2-3 different definitions for buckets for different files, but they are essentially the same; only the props are optional in some files.